crates/nu-parser/src/parser.rs

```diff
@@ -1911,8 +1911,16 @@ pub fn parse_range(working_set: &mut StateWorkingSet, span: Span) -> Option<Expr
         return None;
     }
 
-    // First, figure out what exact operators are used and determine their positions
-    let dotdot_pos: Vec<_> = token.match_indices("..").map(|(pos, _)| pos).collect();
+    let dotdot_pos: Vec<_> = token
+        .match_indices("..")
+        .filter_map(|(pos, _)| {
+            // paren_depth = count of unclosed parens prior to pos
+            let before = &token[..pos];
+            let paren_depth = before.chars().filter(|&c| c == '(').count()
+                - before.chars().filter(|&c| c == ')').count();
+            if paren_depth == 0 { Some(pos) } else { None }
+        })
+        .collect();
```

Comment:
The complexity is not ideal, we can easily make it O(n) by a stateful iteration.

Reply:
Forget about it, it probably won't hurt much considering `..` is not a common pattern.